File: 2021 marzec/python/1.py
with open("galerie.txt", "r") as f:
    galerie = f.read().strip().split('\n')

kraj, miasto, lokal = [], [], []
for g in galerie:
    g = g.split()
    kraj.append(g[0])
    miasto.append(g[1])
    lokal.append(list(map(int, g[2:])))

d = dict()
for k in kraj:
    if k not in d.keys():
        d[k] = 0
    d[k] += 1

with open("wynik4_1.txt", "w") as f:
    for k, i in d.items():
        f.write(str(k) + ' ' + str(i) + '\n')

File: 2021 marzec/c++/1.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <map>
using namespace std;

string kraj[50], miasto[50];
int sklep[50][140];
map <string, int> m;

int main()
{
    ifstream in("galerie.txt");
    ofstream out("wynik4_1.txt");
    
    for (int i = 0; i < 50; i++)
    {
        in >> kraj[i] >> miasto[i];
        for (int j = 0; j < 140; j++)
            in >> sklep[i][j];
    }

    for (int i = 0; i < 50; i++)
        m[kraj[i]]++;
    
    for (pair <string, int> x: m)
        out << x.first << ' ' << x.second << '\n';
}